### ISSUE TITLE
Revert antenna correction for NOAA-15 AMSU-A

### ISSUE THREAD

**Issue by ADCollard** [NOAA-EMC/GSI#380]
After some digging into our AMSU-A preprocessing, it appears that the assumption that the AMSU-A dump files contain antenna temperatures is not 100% true.  All currently assimilated AMSU-A dump files are antenna temperatures, except NOAA-15 (and NOAA-16 which is no longer functional).

We need to put some logic in GSI to ensure that NOAA-15 (and -16) are not antenna-corrected.

**Comment by ADCollard** [comment]
Some background:

The antenna correction is actually performed inside the [tranamsua](https://github.com/NOAA-EMC/satingest/blob/develop/sorc/bufr_tranamsua.fd/tranamsua.f) code.  It reads in a coefficient file and spits out corrected or uncorrected files depending on the setting of PROCESS_TA and PROCESS_TB.  For almost every AMSU-A, the coefficient file is missing.

From: /gpfs/dell1/nco/ops/com/output/prod/20220509/itovs_2359.o91794409:

   compress='YES',     ! BUFR compression switch
  coefile='/dev/null',       ! Path to input coefficient file
  process_Tb='NO', ! Process bright. temps into BUFR?
  process_Ta='YES'  ! Process antenna temps into BUFR? Ta to Tb conversion coefficient file missing,

and further down the same file:

 Ta to Tb conversion coefficient file missing,
 use default values --> no Ta to Tb conversion
 Eta factor not found in coefficient file, use
 default values --> no Ta to Tb conversion

So the Tb file should contain uncorrected (Ta) values in this case.

But for NOAA-15 (and NOAA-16 which is no longer active) the coefficient file does exist and is used and the antenna correction IS being applied.  So for NOAA-15 only the tank file xx023 contains brightness temperatures, but antenna temperatures for everything else.  I note that NOAA-15 and -16 were launched before the tranamsua code was written.

**Comment by HaixiaLiu-NOAA** [comment]
@ADCollard Thank you for documenting the issue with background details. @ScottSieron-NOAA As we discussed during the tag-up meeting, please work on the code modification then update your progress here to let us know once it is ready to review the code changes. Thank you

**Comment by ScottSieron-NOAA** [comment]
On Monday my GSI was rebased and transitioned to the now-called develop branch. From there a branch called n15_TaTb was created and the code changes were implemented. It is straightforward change:

if ( (amsua .or. amsub .or. mhs) **.and. &
  .not.(jsatid == 'n15' .or. jsatid == 'n16')** )then
    ...
    call apply_antcorr(...)
    ...
end if

The logic was tested with the use of elseif and else statements and write statements.

A single cycle experiment was performed for the two setups, one with n15 antenna correction, and one with the new exclusion of n15 antenna correction.

The images are histograms of OminusF (no bias correction) of amsua n15 channel 8. The first is with the original code in which n15 observations had the antenna correction erroneously applied.
![image](https://user-images.githubusercontent.com/70583461/169580827-e4699007-71fc-41b4-8002-25ee06a276a9.png)

The second is with the above change which excludes n15 from antenna correction.
![image](https://user-images.githubusercontent.com/70583461/169580171-f901574b-f51e-41a1-82ed-35309a9a6b9e.png)

The two histograms differ indicating the efficacy of the change in antenna correction. The histogram with n15 excluded from antenna correction is more centered at 0, further indicating that the change is appropriate.

**Comment by ScottSieron-NOAA** [comment]
The authoritative repository update on the afternoon of Wednesday was merged into my GSI.

I am running into difficulty running the GSI after merging this update. I discovered changes required in the single-cycle analysis scripts, e.g., /scratch1/NCEPDEV/da/Scott.Sieron/GDAS/scripts/rungsi_amsua_n15_noTaTb.sh . After merging the update and installing, I changed the module load to gsi_hera.intel.lua . I also changed the execs from exec/global_gsi.x and exec/ncdiag_cat.x , to install/bin/gsi.x and install/bin/nc_diag_cat.x , respectively.

The sbatch job runs for the full 30 minutes instead of ~3 minutes like before, and what I need--the file diag_amsua_n15_ges.2020100300.nc4 and diag_amsua_n18_ges.2020100300.nc4--is empty and is nonexistent, respectively. On line 3990 of /scratch1/NCEPDEV/da/Scott.Sieron/GDAS/scripts/gsi766.o31747268 , there is a segmentation fault. In /scratch2/NCEPDEV/stmp1/Scott.Sieron/tmp382/gsi-test-n15-noTaTb.2020100300 , stdout has an error I don't recall seeing before: BUFRLIB: ISETPRM - DYNAMIC MEMORY ALLOCATION IS NOT SUPPORTED FOR THIS BUILD OF BUFRLIB.

I have a response from Michael Lueken on how to best proceed from here.

**Comment by ADCollard** [comment]
@ScottSieron-NOAA Did you get anywhere with this?  I assume you are using @emilyhcliu's scripts?  I just ran develop using Emily's scripts and was successful.  It looks like it's the ncdiag cat step that is causing you trouble.  I pointed to install/bin/ncdiag_cat_serial.x not install/bin/nc_diag_cat.x  though.

**Comment by HaixiaLiu-NOAA** [comment]
@ScottSieron-NOAA What is the reason to change from using exec to install executables? After bringing the authoritative updates to your own fork, you can still use the executables in the exec directory after recompiling. And are the changes required in the single-cycle analysis scripts referring to point to different gsi executable or more than that?

**Comment by ScottSieron-NOAA** [comment]
@ADCollard Mike suggested changing the bufr version, but this did not resolve the issue (except for the BUFRLIB: ISETPRM - DYNAMIC MEMORY ALLOCATION IS NOT SUPPORTED FOR THIS BUILD OF BUFRLIB).

But your suggestion of instead using ncdiag_cat_serial.x has resolved the issue, thank you.

@HaixiaLiu-NOAA I suppose I am doing something wrong with the compiling because an exec directory is not created for me. 

The only other change I've made to Emily's GMI scripts besides changing which sensors to assimilating, loading gsi_hera.intel.lua instead of modulefile.ProdGSI.hera, and changing the gsiexec and catexec, is setting fixgsi=/scratch1/NCEPDEV/da/Scott.Sieron/GDAS/GSI/fix , which I did because the fix files in develop have now been updated.

**Comment by MichaelLueken** [comment]
> @ScottSieron-NOAA What is the reason to change from using exec to install executables? After bringing the authoritative updates to your own fork, you can still use the executables in the exec directory after recompiling. And are the changes required in the single-cycle analysis scripts referring to point to different gsi executable or more than that?

@HaixiaLiu-NOAA That shouldn't be happening.  Rahul changed the build so that all executables end up in install/bin.  The moving of executables from build/bin to exec has been removed.  Once the HEAD of the authoritative develop branch has been merged to your fork, there should no longer be an exec directory with executables.  If there is, then something went wrong with the merge.

**Comment by rtodling** [comment]
Andrew and Emily: Given that you are changing GSI to accommodate the pre-processing error, I take it that there is, and neither there will be, a plan to fix the pre-processing. Am I right? It would be rather hard to know when to apply the fix in GSI or not ... although I supposed a date can also be patched in the GSI code for when to treat as antenna T or or ... 

Please let me know.
Cheers.

**Comment by ADCollard** [comment]
@rtodling Management made the decision to not fix the preprocessing so that we could maintain continuity in the dump files (and not re-process 25 years of AMSU data).   It is unclear for how much longer NOAA-15 will be around (launched in 1998!) and actually the bias correction does a very good job of mimicking the antenna correction anyway.  Just to be clear the double-correction of NOAA-15 did not make it into operations (the upgrade is scheduled for November).
